webrisk/database_test.go

```diff
@@ -0,0 +1,753 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webrisk
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+	"fmt"
+
+	pb "github.com/GoogleCloudPlatform/golang-samples/webrisk/internal/webrisk_proto"
+	pt "github.com/golang/protobuf/ptypes"
+)
+
+func mustGetTempFile(t *testing.T) string {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	path := f.Name()
+	if err := f.Close(); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	return path
+}
+
+func mustDecodeHex(t *testing.T, s string) []byte {
+	buf, err := hex.DecodeString(s)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	return buf
+}
+
+func newHashSet(phs hashPrefixes) (hs hashSet) {
+	hs.Import(phs)
+	return hs
+}
+
+func TestDatabaseInit(t *testing.T) {
+	path := mustGetTempFile(t)
+	defer os.Remove(path)
+
+	now := time.Unix(1451436338, 951473000)
+	mockNow := func() time.Time { return now }
+
+	vectors := []struct {
+		config *Config   // Input configuration
+		oldDB  *database // The old database (before export)
+		newDB  *database // The expected new database (after import)
+		fail   bool      // Expected failure
+	}{{
+		// Load from a valid database file.
+		config: &Config{
+			ThreatLists: []ThreatType{
+				0,
```

Comment:
Use `pb.ThreatType_THREAT_TYPE_UNSPECIFIED` and friends throughout?

Reply:
Done.